homeassistant/components/recorder/migration.py

```diff
@@ -1288,6 +1293,55 @@ def migrate_context_ids(instance: Recorder) -> bool:
     return is_done
 
 
+def migrate_event_type_ids(instance: Recorder) -> bool:
+    """Migrate event_type to event_type_ids."""
+    session_maker = instance.get_session
+    _LOGGER.debug("Migrating event_types")
+    event_type_manager = instance.event_type_manager
+    with session_scope(session=session_maker()) as session:
+        if events := session.execute(find_event_type_to_migrate()).all():
+            event_types = {event_type for _, event_type in events}
+            event_type_to_id = event_type_manager.get_many(event_types, session)
+            if missing_event_types := {
+                event_type
+                for event_type, event_id in event_type_to_id.items()
+                if event_id is None
+            }:
+                missing_db_event_types = [
+                    EventTypes(event_type=event_type)
+                    for event_type in missing_event_types
+                ]
+                session.add_all(missing_db_event_types)
+                session.flush()  # Assign ids
+                for db_event_type in missing_db_event_types:
+                    assert db_event_type.event_type is not None
```

Comment:
We should tell the `event_type_manager` about them so it doesn't have to reselect

Reply:
maybe better to leave it as is in case the transaction fails